apps/opik-backend/src/main/java/com/comet/opik/domain/attachment/AttachmentStripperService.java

```diff
@@ -0,0 +1,573 @@
+        this.attachmentService = attachmentService;
+        this.idGenerator = idGenerator;
+        this.objectMapper = objectMapper;
+        this.s3Config = opikConfig.getS3Config();
+
+        // Initialize OpenTelemetry metrics using global instance
+        Meter meter = GlobalOpenTelemetry.get().getMeter("opik.attachments");
+
+        this.attachmentsProcessed = meter
+                .counterBuilder("opik.attachments.processed")
+                .setDescription("Number of attachments successfully processed and uploaded")
+                .build();
+        this.attachmentsSkipped = meter
+                .counterBuilder("opik.attachments.skipped")
+                .setDescription("Number of base64 strings skipped (not valid attachments)")
+                .build();
+        this.attachmentErrors = meter
+                .counterBuilder("opik.attachments.errors")
+                .setDescription("Number of errors during attachment processing")
+                .build();
+        this.processingTimer = meter
+                .histogramBuilder("opik.attachments.processing.duration.ms")
+                .setDescription("Time spent processing attachments in milliseconds")
+                .ofLongs()
+                .build();
+
+        // Compile the regex pattern once during construction based on configuration
+        int minLength = s3Config.getStripAttachmentsMinSize();
+        this.minBase64Size = minLength;
+        this.base64Pattern = Pattern.compile("([A-Za-z0-9+/]{" + minLength + ",}={0,2})");
+
+        log.info("AttachmentStripperService initialized with minBase64Length: {}", minLength);
+    }
+
+    /**
+     * Strips attachments from a Trace entity.
+     *
+     * TODO: This one and the similar ones below should be refactored to use the same method if
+     * we can make all these types to share the same interface (shouldn't they?)
+     */
+    public Mono<Trace> stripAttachmentsFromTrace(Trace trace, String workspaceId, String userName, String projectName) {
+        return Mono.fromCallable(() -> {
+            Trace.TraceBuilder builder = trace.toBuilder();
+
+            if (trace.input() != null) {
+                JsonNode processedInput = stripAttachments(trace.input(), trace.id(), EntityType.TRACE,
+                        workspaceId, userName, projectName, "input");
+                builder.input(processedInput);
+            }
+
+            if (trace.output() != null) {
+                JsonNode processedOutput = stripAttachments(trace.output(), trace.id(), EntityType.TRACE,
+                        workspaceId, userName, projectName, "output");
+                builder.output(processedOutput);
+            }
+
+            if (trace.metadata() != null) {
+                JsonNode processedMetadata = stripAttachments(trace.metadata(), trace.id(), EntityType.TRACE,
+                        workspaceId, userName, projectName, "metadata");
+                builder.metadata(processedMetadata);
+            }
```

Comment:
We can process each of those async and wait for all to finish to rebuild the entity